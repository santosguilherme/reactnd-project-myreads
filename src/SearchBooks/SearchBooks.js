import React, {Component} from 'react';
import PropTypes from 'prop-types';

import {Link} from 'react-router-dom';

import If from '../commons/components/If';
import Book from '../Book/Book';

import './SearchBooks.css';


class SearchBooks extends Component {
    state = {
        query: ''
    };

    handleChangeSearchInput = event => {
        const {onSearchBooks} = this.props;
        const query = event.target.value;

        this.setState({query});
        onSearchBooks(query);
    };

    render() {
        const {query} = this.state;
        const {onUpdateBook, message, booksResult = []} = this.props;

        return (
            <div className="search-books">
                <div className="search-books-bar">
                    <Link
                        className="close-search"
                        to="/"
                    >
                        Close
                    </Link>
                    <div className="search-books-input-wrapper">
                        <input
                            type="text"
                            placeholder="Search by title or author"
                            value={query}
                            onChange={this.handleChangeSearchInput}
                            autoFocus={true}
                        />
                    </div>
                </div>
                <div className="search-books-results">
                    <If test={message}>
                        <p align="center">{message}</p>
                    </If>
                    <ol className="books-grid">
                        {booksResult.map((book, index) => (
                            <li key={index}>
                                <Book
                                    book={book}
                                    onUpdateBook={onUpdateBook}
                                />
                            </li>
                        ))}
                    </ol>
                </div>
            </div>
        );
    }
}

SearchBooks.defaultProps = {
    message: 'Enter the title or actor you want to search, in the field above.'
};

SearchBooks.propTypes = {
    booksResult: PropTypes.array,
    message: PropTypes.string,
    onSearchBooks: PropTypes.func.isRequired,
    onUpdateBook: PropTypes.func.isRequired
};

export default SearchBooks;