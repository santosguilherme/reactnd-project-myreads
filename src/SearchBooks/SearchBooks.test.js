import React from 'react';
import {shallow} from 'enzyme';

import SearchBooks from './SearchBooks';

it('renders without crashing', () => {
    const props = {
        booksResult: [{
            title: 'My Book',
            shelf: 'read',
            authors: ['Santos, Guilherme']
        }],
        message: '',
        onSearchBooks: jest.fn(),
        onUpdateBook: jest.fn()
    };
    const wrapper = shallow(<SearchBooks {...props}/>);

    expect(wrapper).toMatchSnapshot();
});

it('renders with empty array', () => {
    const props = {
        booksResult: [],
        message: '',
        onSearchBooks: jest.fn(),
        onUpdateBook: jest.fn()
    };
    const wrapper = shallow(<SearchBooks {...props}/>);

    expect(wrapper).toMatchSnapshot();
});

it('Should call the onSearchBooks property when change input', () => {
    const props = {
        booksResult: [],
        message: '',
        onSearchBooks: jest.fn(),
        onUpdateBook: jest.fn()
    };
    const wrapper = shallow(<SearchBooks {...props}/>);
    const input = wrapper.find('input');

    input.simulate('change', {target: {value: 'something'}});

    expect(props.onSearchBooks).toHaveBeenCalledTimes(1);
    expect(props.onSearchBooks.mock.calls[0]).toEqual(['something']);
});