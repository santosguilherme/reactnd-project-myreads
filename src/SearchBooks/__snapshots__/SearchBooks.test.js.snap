// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with empty array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBooks
    booksResult={Array []}
    message=""
    onSearchBooks={[Function]}
    onUpdateBook={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-books-bar"
>
          <Link
                    className="close-search"
                    onClick={undefined}
                    replace={false}
                    to="/"
          >
                    Close
          </Link>
          <div
                    className="search-books-input-wrapper"
          >
                    <Debounce
                              handler="onChange"
                              handlers={Array []}
                              time="200"
                    >
                              <input
                                        autoFocus={true}
                                        onChange={[Function]}
                                        placeholder="Search by title or author"
                                        type="text"
                              />
                    </Debounce>
          </div>
</div>,
        <div
          className="search-books-results"
>
          <If
                    test=""
          >
                    <p
                              align="center"
                    />
          </If>
          <ol
                    className="books-grid"
          >
                    
          </ol>
</div>,
      ],
      "className": "search-books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="close-search"
              onClick={undefined}
              replace={false}
              to="/"
>
              Close
</Link>,
            <div
              className="search-books-input-wrapper"
>
              <Debounce
                            handler="onChange"
                            handlers={Array []}
                            time="200"
              >
                            <input
                                          autoFocus={true}
                                          onChange={[Function]}
                                          placeholder="Search by title or author"
                                          type="text"
                            />
              </Debounce>
</div>,
          ],
          "className": "search-books-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "className": "close-search",
              "onClick": undefined,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <Debounce
                handler="onChange"
                handlers={Array []}
                time="200"
>
                <input
                                autoFocus={true}
                                onChange={[Function]}
                                placeholder="Search by title or author"
                                type="text"
                />
</Debounce>,
              "className": "search-books-input-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <input
                  autoFocus={true}
                  onChange={[Function]}
                  placeholder="Search by title or author"
                  type="text"
/>,
                "handler": "onChange",
                "handlers": Array [],
                "time": "200",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "autoFocus": true,
                  "onChange": [Function],
                  "placeholder": "Search by title or author",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <If
              test=""
>
              <p
                            align="center"
              />
</If>,
            <ol
              className="books-grid"
>
              
</ol>,
          ],
          "className": "search-books-results",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <p
                align="center"
/>,
              "test": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "align": "center",
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "books-grid",
            },
            "ref": null,
            "rendered": Array [],
            "type": "ol",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-books-bar"
>
            <Link
                        className="close-search"
                        onClick={undefined}
                        replace={false}
                        to="/"
            >
                        Close
            </Link>
            <div
                        className="search-books-input-wrapper"
            >
                        <Debounce
                                    handler="onChange"
                                    handlers={Array []}
                                    time="200"
                        >
                                    <input
                                                autoFocus={true}
                                                onChange={[Function]}
                                                placeholder="Search by title or author"
                                                type="text"
                                    />
                        </Debounce>
            </div>
</div>,
          <div
            className="search-books-results"
>
            <If
                        test=""
            >
                        <p
                                    align="center"
                        />
            </If>
            <ol
                        className="books-grid"
            >
                        
            </ol>
</div>,
        ],
        "className": "search-books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="close-search"
                onClick={undefined}
                replace={false}
                to="/"
>
                Close
</Link>,
              <div
                className="search-books-input-wrapper"
>
                <Debounce
                                handler="onChange"
                                handlers={Array []}
                                time="200"
                >
                                <input
                                                autoFocus={true}
                                                onChange={[Function]}
                                                placeholder="Search by title or author"
                                                type="text"
                                />
                </Debounce>
</div>,
            ],
            "className": "search-books-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Close",
                "className": "close-search",
                "onClick": undefined,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="200"
>
                  <input
                                    autoFocus={true}
                                    onChange={[Function]}
                                    placeholder="Search by title or author"
                                    type="text"
                  />
</Debounce>,
                "className": "search-books-input-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <input
                    autoFocus={true}
                    onChange={[Function]}
                    placeholder="Search by title or author"
                    type="text"
/>,
                  "handler": "onChange",
                  "handlers": Array [],
                  "time": "200",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "autoFocus": true,
                    "onChange": [Function],
                    "placeholder": "Search by title or author",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <If
                test=""
>
                <p
                                align="center"
                />
</If>,
              <ol
                className="books-grid"
>
                
</ol>,
            ],
            "className": "search-books-results",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": <p
                  align="center"
/>,
                "test": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "align": "center",
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "books-grid",
              },
              "ref": null,
              "rendered": Array [],
              "type": "ol",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBooks
    booksResult={
        Array [
            Object {
              "authors": Array [
                "Santos, Guilherme",
              ],
              "shelf": "read",
              "title": "My Book",
            },
          ]
    }
    message=""
    onSearchBooks={[Function]}
    onUpdateBook={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="search-books-bar"
>
          <Link
                    className="close-search"
                    onClick={undefined}
                    replace={false}
                    to="/"
          >
                    Close
          </Link>
          <div
                    className="search-books-input-wrapper"
          >
                    <Debounce
                              handler="onChange"
                              handlers={Array []}
                              time="200"
                    >
                              <input
                                        autoFocus={true}
                                        onChange={[Function]}
                                        placeholder="Search by title or author"
                                        type="text"
                              />
                    </Debounce>
          </div>
</div>,
        <div
          className="search-books-results"
>
          <If
                    test=""
          >
                    <p
                              align="center"
                    />
          </If>
          <ol
                    className="books-grid"
          >
                    <li>
                              <Book
                                        book={
                                                  Object {
                                                            "authors": Array [
                                                              "Santos, Guilherme",
                                                            ],
                                                            "shelf": "read",
                                                            "title": "My Book",
                                                          }
                                        }
                                        onUpdateBook={[Function]}
                              />
                    </li>
          </ol>
</div>,
      ],
      "className": "search-books",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="close-search"
              onClick={undefined}
              replace={false}
              to="/"
>
              Close
</Link>,
            <div
              className="search-books-input-wrapper"
>
              <Debounce
                            handler="onChange"
                            handlers={Array []}
                            time="200"
              >
                            <input
                                          autoFocus={true}
                                          onChange={[Function]}
                                          placeholder="Search by title or author"
                                          type="text"
                            />
              </Debounce>
</div>,
          ],
          "className": "search-books-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "Close",
              "className": "close-search",
              "onClick": undefined,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <Debounce
                handler="onChange"
                handlers={Array []}
                time="200"
>
                <input
                                autoFocus={true}
                                onChange={[Function]}
                                placeholder="Search by title or author"
                                type="text"
                />
</Debounce>,
              "className": "search-books-input-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <input
                  autoFocus={true}
                  onChange={[Function]}
                  placeholder="Search by title or author"
                  type="text"
/>,
                "handler": "onChange",
                "handlers": Array [],
                "time": "200",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "autoFocus": true,
                  "onChange": [Function],
                  "placeholder": "Search by title or author",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <If
              test=""
>
              <p
                            align="center"
              />
</If>,
            <ol
              className="books-grid"
>
              <li>
                            <Book
                                          book={
                                                        Object {
                                                                      "authors": Array [
                                                                        "Santos, Guilherme",
                                                                      ],
                                                                      "shelf": "read",
                                                                      "title": "My Book",
                                                                    }
                                          }
                                          onUpdateBook={[Function]}
                            />
              </li>
</ol>,
          ],
          "className": "search-books-results",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <p
                align="center"
/>,
              "test": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "align": "center",
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  <Book
                                    book={
                                                      Object {
                                                                        "authors": Array [
                                                                          "Santos, Guilherme",
                                                                        ],
                                                                        "shelf": "read",
                                                                        "title": "My Book",
                                                                      }
                                    }
                                    onUpdateBook={[Function]}
                  />
</li>,
              ],
              "className": "books-grid",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": <Book
                    book={
                                        Object {
                                                            "authors": Array [
                                                              "Santos, Guilherme",
                                                            ],
                                                            "shelf": "read",
                                                            "title": "My Book",
                                                          }
                    }
                    onUpdateBook={[Function]}
/>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "book": Object {
                      "authors": Array [
                        "Santos, Guilherme",
                      ],
                      "shelf": "read",
                      "title": "My Book",
                    },
                    "onUpdateBook": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "li",
              },
            ],
            "type": "ol",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="search-books-bar"
>
            <Link
                        className="close-search"
                        onClick={undefined}
                        replace={false}
                        to="/"
            >
                        Close
            </Link>
            <div
                        className="search-books-input-wrapper"
            >
                        <Debounce
                                    handler="onChange"
                                    handlers={Array []}
                                    time="200"
                        >
                                    <input
                                                autoFocus={true}
                                                onChange={[Function]}
                                                placeholder="Search by title or author"
                                                type="text"
                                    />
                        </Debounce>
            </div>
</div>,
          <div
            className="search-books-results"
>
            <If
                        test=""
            >
                        <p
                                    align="center"
                        />
            </If>
            <ol
                        className="books-grid"
            >
                        <li>
                                    <Book
                                                book={
                                                            Object {
                                                                        "authors": Array [
                                                                          "Santos, Guilherme",
                                                                        ],
                                                                        "shelf": "read",
                                                                        "title": "My Book",
                                                                      }
                                                }
                                                onUpdateBook={[Function]}
                                    />
                        </li>
            </ol>
</div>,
        ],
        "className": "search-books",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="close-search"
                onClick={undefined}
                replace={false}
                to="/"
>
                Close
</Link>,
              <div
                className="search-books-input-wrapper"
>
                <Debounce
                                handler="onChange"
                                handlers={Array []}
                                time="200"
                >
                                <input
                                                autoFocus={true}
                                                onChange={[Function]}
                                                placeholder="Search by title or author"
                                                type="text"
                                />
                </Debounce>
</div>,
            ],
            "className": "search-books-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "Close",
                "className": "close-search",
                "onClick": undefined,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": <Debounce
                  handler="onChange"
                  handlers={Array []}
                  time="200"
>
                  <input
                                    autoFocus={true}
                                    onChange={[Function]}
                                    placeholder="Search by title or author"
                                    type="text"
                  />
</Debounce>,
                "className": "search-books-input-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": <input
                    autoFocus={true}
                    onChange={[Function]}
                    placeholder="Search by title or author"
                    type="text"
/>,
                  "handler": "onChange",
                  "handlers": Array [],
                  "time": "200",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "autoFocus": true,
                    "onChange": [Function],
                    "placeholder": "Search by title or author",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <If
                test=""
>
                <p
                                align="center"
                />
</If>,
              <ol
                className="books-grid"
>
                <li>
                                <Book
                                                book={
                                                                Object {
                                                                                "authors": Array [
                                                                                  "Santos, Guilherme",
                                                                                ],
                                                                                "shelf": "read",
                                                                                "title": "My Book",
                                                                              }
                                                }
                                                onUpdateBook={[Function]}
                                />
                </li>
</ol>,
            ],
            "className": "search-books-results",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "children": <p
                  align="center"
/>,
                "test": "",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "align": "center",
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                    <Book
                                        book={
                                                            Object {
                                                                                "authors": Array [
                                                                                  "Santos, Guilherme",
                                                                                ],
                                                                                "shelf": "read",
                                                                                "title": "My Book",
                                                                              }
                                        }
                                        onUpdateBook={[Function]}
                    />
</li>,
                ],
                "className": "books-grid",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": <Book
                      book={
                                            Object {
                                                                  "authors": Array [
                                                                    "Santos, Guilherme",
                                                                  ],
                                                                  "shelf": "read",
                                                                  "title": "My Book",
                                                                }
                      }
                      onUpdateBook={[Function]}
/>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "book": Object {
                        "authors": Array [
                          "Santos, Guilherme",
                        ],
                        "shelf": "read",
                        "title": "My Book",
                      },
                      "onUpdateBook": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "li",
                },
              ],
              "type": "ol",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
